File: code/mathematical_algorithms/src/tribonacci_numbers/tribonacci_numbers.cpp
/* Part of Cosmos by OpenGenus Foundation */

#include <iostream>


using namespace std;

int main()
{
    int n, i;
    int a = 0, b = 1, c = 2, d;
    cout << "Enter a number\n";
    cin >> n;
    cout << a << " " << b << " " << c << " ";
    for (i = 0; i < n - 3; i++)
    {
        d = a + b + c;
        cout << d << " ";
        a = b;
        b = c;
        c = d;

    }
    return 0;
}

File: code/mathematical-algorithms/tribonacci_numbers/tribonacci_numbers.js
/* Part of Cosmos by OpenGenus Foundation */

const readline = require('readline');

const ioInterface = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

function tribonacii(number) {
    let a = 0,
        b = 0,
        c = 1,
        d;
    for (let i = 0; i < number - 3; i++) {
        d = a + b + c;
        console.log(`${d} `);
        a = b;
        b = c;
        c = d;
    }
}

ioInterface.question('Enter a number: ', (number) => {
    tribonacii(parseInt(number));
    ioInterface.close();
});
